### PR TITLE
Add unread-class to category children

### PR DESCRIPTION
**Pull request by aStonedPenguin**
Useful for themes.

ex:
example

### CODE REVIEW

File: src/categories.js [hunk 1]
```diff
@@ -173,6 +174,23 @@ Categories.getTagWhitelist = function (cids, callback) {
 	db.getSortedSetsMembers(keys, callback);
 };
 
+function calculateUnreadChildren(children, uid) {
+	var cids = children.map(function (child) {
+		return child.cid;
+	});
+
+	Categories.hasReadCategories(cids, uid, function (err, data) {
```
Comment (barisusakli) [inline src/categories.js]: This won't work as you expect it to, you are calling an async function inside a synchronous forEach so the execution will just keep going and won't wait for the results of hasReadCategories.

Comment (barisusakli) [reply src/categories.js]: calculateUnreadChildren needs a callback parameter as last param and needs to be moved out of the .forEach

File: src/categories.js [hunk 2]
```diff
@@ -173,6 +174,23 @@ Categories.getTagWhitelist = function (cids, callback) {
...
+		if (err) {
+			return;
```
Comment (barisusakli) [inline src/categories.js]: send the error to the newly added callback as first param

File: src/categories.js (hunk 1 above)
Comment (aStonedPenguin) [reply src/categories.js]: Worked fine when I tested it. Will fix it in a moment.

File: src/categories.js [hunk 3]
```diff
@@ -259,9 +259,29 @@ function getChildrenRecursive(category, uid, callback) {
...
+		function (children, next) {
+			children = children.filter(Boolean);
+			category.children = children;
+
+			async.map(children, function (child, next) {
+				next(null, child.cid);
+			}, next);
+		},
+		function (cids, next) {
+			Categories.hasReadCategories(cids, uid, function (err, data) {
```
Comment (barisusakli) [inline src/categories.js]: there is no need to use async.map here.
You can rewrite it as below
```
var cids = children.map(function (child) {
    return child.cid;
});
Categories.hasReadCategories(cids, uid, next);
```

Comment (aStonedPenguin) [reply src/categories.js]: done

**Pull request merged**